profiling/src/sapi.rs

```diff
@@ -0,0 +1,92 @@
+use std::collections::HashMap;
+use std::mem::MaybeUninit;
+use std::sync::Once;
+
+// todo: unify with ../component/sapi
+#[derive(Copy, Clone, Eq, PartialEq)]
+#[repr(C)]
+pub enum Sapi {
+    Unknown = 0,
+    Apache2Handler,
+    CgiFcgi,
+    Cli,
+    CliServer,
+    Embed,
+    FpmFcgi,
+    Litespeed,
+    PhpDbg,
+    Tea,
+}
+
+impl Sapi {
+    pub fn from_name(name: &str) -> Sapi {
+        static mut SAPIS: MaybeUninit<HashMap<&str, Sapi>> = MaybeUninit::uninit();
+        static ONCE: Once = Once::new();
+        ONCE.call_once(|| {
+            let sapis = HashMap::from([
+                ("apache2handler", Sapi::Apache2Handler),
+                ("cgi-fcgi", Sapi::CgiFcgi),
+                ("cli", Sapi::Cli),
+                ("cli-server", Sapi::CliServer),
+                ("embed", Sapi::Embed),
+                ("fpm-fcgi", Sapi::FpmFcgi),
+                ("litespeed", Sapi::Litespeed),
+                ("phpdbg", Sapi::PhpDbg),
+                ("tea", Sapi::Tea),
+            ]);
+            unsafe { SAPIS.write(sapis) };
+        });
```

Comment:
This seems like a job for the `maplit` crate.

Reply:
I think before Rust 1.56 that `maplit` would be nice, but since then we have `from` and `from_iter` for `HashMap`. Using `maplit` would be changing out the `(_, _),` structure for `_ => _,` via macro, and I'm not sure how that's better? If it could do it in const contexts that would be awesome, but it can't. Am I missing something?